nexus/reconfigurator/planning/src/example.rs

```diff
@@ -1103,6 +1121,299 @@ mod tests {
+        }
+
+        // Query with the qorb DNS resolver.
+        {
+            let resolver =
+                QorbResolver::new(vec![dns.dns_server.local_address()]);
+
+            // Ensure that service names can be looked up via the qorb resolver.
+            let mut services_with_errors = BTreeMap::new();
+            for (service, expected_result) in service_names_to_query(&blueprint)
+            {
+                // We can't really use qorb to query error cases, since those
+                // are handled internally by the resolver. Just query the
+                // success cases.
+                if expected_result.is_err() {
+                    continue;
+                }
+
+                eprintln!("*** using qorb to look up DNS for {:?}", service);
+                let mut srv_resolver = resolver.for_service(service);
+                let mut monitor_rx = srv_resolver.monitor();
+
+                // Wait for at least one result to be returned. We don't want to
+                // wait forever, so set a reasonable timeout. (15s matches the
+                // internal timeout within the simple DNS resolver.)
+                let backends = match tokio::time::timeout(
+                    Duration::from_secs(15),
+                    monitor_rx.wait_for(|backends| !backends.is_empty()),
+                )
+                .await
+                {
+                    Ok(Ok(backends)) => backends,
+                    Ok(Err(e)) => {
+                        services_with_errors.insert(service, anyhow!(e));
+                        continue;
+                    }
+                    Err(e) => {
+                        services_with_errors.insert(service, anyhow!(e));
+                        continue;
+                    }
+                };
+
+                eprintln!("*** qorb lookup successful: {:?}", &**backends);
+            }
+        }
+
+        logctx.cleanup_successful();
+    }
+
+    /// Returns the list of potential DNS service names in an example system,
+    /// along with whether we expect a successful or error response for each of
+    /// them.
+    fn service_names_to_query(
+        blueprint: &Blueprint,
+    ) -> BTreeMap<ServiceName, Result<(), QueryError>> {
+        let mut out = BTreeMap::new();
+
+        for service in ServiceName::iter() {
+            match service {
+                // Services that exist in the example system.
+                ServiceName::Clickhouse
+                | ServiceName::ClickhouseAdminSingleServer
+                | ServiceName::ClickhouseNative
+                | ServiceName::InternalDns
+                | ServiceName::Nexus
+                | ServiceName::NexusLockstep
+                | ServiceName::OximeterReader
+                | ServiceName::RepoDepot
+                | ServiceName::CruciblePantry => {
+                    out.insert(service, Ok(()));
+                }
+                // Services that are not currently part of the example system.
```

Comment:
(sorry if we already did this)
Can you file an issue for including these in the example system?

Reply:
#9349